frontend/blog-angular-ui/src/app/login/login.component.spec.ts

```
@@ -1,31 +1,109 @@
     .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(LoginComponent);
     component = fixture.componentInstance;
+    router = TestBed.inject(Router);
+    hardcodedAuthService = TestBed.inject(HardcodedAuthenticationService);
+    basicAuthService = TestBed.inject(BasicAuthenticationService);
+    httpTestingController = TestBed.inject(HttpTestingController);
     fixture.detectChanges();
   });
```

Biome (1.9.4)

[error] Disallow duplicate setup and teardown hooks.

Disallow beforeEach duplicacy inside the describe function.

(lint/suspicious/noDuplicateTestHooks)